include/net/lwm2m.h

```diff
@@ -5,20 +5,34 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+/** @file lwm2m.h
+ *
+ * @defgroup lwm2m_api LwM2M high-level API
+ * @ingroup networking
+ * @{
+ * @brief LwM2M high-level API
+ *
+ * @details
+ * LwM2M high-level interface is defined in this header.
+ *
+ * @note The implementation assumes UDP module is enabled.
+ *
+ * @note LwM2M 1.0.x is currently the only supported version.
+ */
+
 #ifndef ZEPHYR_INCLUDE_NET_LWM2M_H_
 #define ZEPHYR_INCLUDE_NET_LWM2M_H_
 
 #include <kernel.h>
 #include <net/coap.h>
 
 /**
- * @brief LwM2M high-level API
- * @defgroup lwm2m_api LwM2M high-level API
- * @{
+ * @brief LwM2M Objects managed by OMA for LwM2M tech specification.  Objects
```

Comment:
As written, this description is picked up as a description of the LWM2M_OBJECT_SECURITY_ID define immediately after this doxygen comment, but I suspect your intention was to describe a group of eight defines below.  That would need to be done with a defgroup around these eight defines. Alas, the Sphinx/Breath system doesn't display nested groups when you use a doxygengroup directive (the nested groups would need a separate doxygengroup directive).  Sigh.  I'd recommend leaving this as you've written for now.